protocol/java/org/openyolo/protocol/Credential.java

```diff
@@ -80,29 +148,25 @@ private Credential(@NonNull Protobufs.Credential proto) {
      */
     @NonNull
     public String getIdentifier() {
-        return mProto.getId();
+        return mId;
     }
 
     /**
-     * The authentication domain against which this credential can be used, if specified.
+     * The authentication method for the credential, which describes the mechanism by which
+     * it should be verified by the receiving app. This may be any valid URI, but is typically
+     * one of the {@link AuthenticationMethods standard values}.
      */
-    @Nullable
-    public AuthenticationDomain getAuthenticationDomain() {
-        String authDomainStr = nullifyEmptyString(mProto.getAuthDomain());
-        if (authDomainStr == null) {
-            return null;
-        }
-
-        return new AuthenticationDomain(mProto.getAuthDomain());
+    @NonNull
+    public AuthenticationMethod getAuthenticationMethod() {
+        return mAuthMethod;
     }
 
     /**
-     * The credential password, if defined. Must be non-empty, but any further restrictions are
-     * the responsibility of the authentication domain to define and validate.
+     * The authentication domain against which this credential can be used, if specified.
```

Comment:
Nit: Remove "if specified".  The authentication domain is guaranteed to be present